src/libraries/Common/src/System/Net/Sockets/SocketConnection.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.IO.Pipelines;
+using System.Net.Sockets;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Connections
+{
+    internal sealed class SocketConnection : Connection, IConnectionProperties
+    {
+        private readonly Socket _socket;
+        private readonly SocketsConnectionFactory? _factory;
+        private readonly IConnectionProperties? _options;
+        private Stream? _stream;
+#if SYSTEM_NET_SOCKETS_DLL
+        private IDuplexPipe? _pipe;
+#endif
+        public override EndPoint? RemoteEndPoint => _socket.RemoteEndPoint;
+        public override EndPoint? LocalEndPoint => _socket.LocalEndPoint;
+        public override IConnectionProperties ConnectionProperties => this;
+
+        public SocketConnection(Socket socket, SocketsConnectionFactory? factory, IConnectionProperties? options)
```

Comment:
I don't see any public API for SocketConnection in the approved API (#40044), including constructors. Why is this public?

Reply:
The class is internal therefore it's not actually `public`. Personally I prefer to define members that should be visible outside the class `public` even on internal classes.  Do we a rule/preference for this in the coding guideline?

Reply:
No, that's fine, I just missed that the class is internal :)